[octree] OctreePointCloudDensity cannot be used

**Context**

This program does not link:
```
#include <pcl/octree/octree_pointcloud_density.h>

using namespace pcl;

int main(int argc, char *argv[]) {
  octree::OctreePointCloudDensity<PointXYZ> octree(0.1);
  return 0;
}
```

**Expected behavior**

Program using `OctreePointCloudDensity` should link correctly.

**Current Behavior**

```
Undefined symbols for architecture x86_64:
  "pcl::octree::OctreeBase<pcl::octree::OctreePointCloudDensityContainer, pcl::octree::OctreeContainerEmpty>::~OctreeBase()", referenced from:
      pcl::octree::OctreePointCloud<pcl::PointXYZ, pcl::octree::OctreePointCloudDensityContainer, pcl::octree::OctreeContainerEmpty, pcl::octree::OctreeBase<pcl::octree::OctreePointCloudDensityContainer, pcl::octree::OctreeContainerEmpty> >::~OctreePointCloud() in wtf.cpp.o
  "pcl::octree::OctreePointCloud<pcl::PointXYZ, pcl::octree::OctreePointCloudDensityContainer, pcl::octree::OctreeContainerEmpty, pcl::octree::OctreeBase<pcl::octree::OctreePointCloudDensityContainer, pcl::octree::OctreeContainerEmpty> >::addPointIdx(int)", referenced from:
      vtable for pcl::octree::OctreePointCloudDensity<pcl::PointXYZ, pcl::octree::OctreePointCloudDensityContainer, pcl::octree::OctreeContainerEmpty> in wtf.cpp.o
      vtable for pcl::octree::OctreePointCloud<pcl::PointXYZ, pcl::octree::OctreePointCloudDensityContainer, pcl::octree::OctreeContainerEmpty, pcl::octree::OctreeBase<pcl::octree::OctreePointCloudDensityContainer, pcl::octree::OctreeContainerEmpty> > in wtf.cpp.o
  "pcl::octree::OctreePointCloud<pcl::PointXYZ, pcl::octree::OctreePointCloudDensityContainer, pcl::octree::OctreeContainerEmpty, pcl::octree::OctreeBase<pcl::octree::OctreePointCloudDensityContainer, pcl::octree::OctreeContainerEmpty> >::addPointsFromInputCloud()", referenced from:
      _main in wtf.cpp.o
  "pcl::octree::OctreePointCloud<pcl::PointXYZ, pcl::octree::OctreePointCloudDensityContainer, pcl::octree::OctreeContainerEmpty, pcl::octree::OctreeBase<pcl::octree::OctreePointCloudDensityContainer, pcl::octree::OctreeContainerEmpty> >::OctreePointCloud(double)", referenced from:
      pcl::octree::OctreePointCloudDensity<pcl::PointXYZ, pcl::octree::OctreePointCloudDensityContainer, pcl::octree::OctreeContainerEmpty>::OctreePointCloudDensity(double) in wtf.cpp.o
  "pcl::octree::OctreePointCloud<pcl::PointXYZ, pcl::octree::OctreePointCloudDensityContainer, pcl::octree::OctreeContainerEmpty, pcl::octree::OctreeBase<pcl::octree::OctreePointCloudDensityContainer, pcl::octree::OctreeContainerEmpty> >::genOctreeKeyForDataT(int const&, pcl::octree::OctreeKey&) const", referenced from:
      vtable for pcl::octree::OctreePointCloudDensity<pcl::PointXYZ, pcl::octree::OctreePointCloudDensityContainer, pcl::octree::OctreeContainerEmpty> in wtf.cpp.o
      vtable for pcl::octree::OctreePointCloud<pcl::PointXYZ, pcl::octree::OctreePointCloudDensityContainer, pcl::octree::OctreeContainerEmpty, pcl::octree::OctreeBase<pcl::octree::OctreePointCloudDensityContainer, pcl::octree::OctreeContainerEmpty> > in wtf.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**To Reproduce**

```
cmake_minimum_required(VERSION 3.18)
project(pcl-wtf)
find_package(PCL REQUIRED)
include_directories(${PCL_INCLUDE_DIRS})
link_libraries(${PCL_LIBRARIES})
add_executable(wtf wtf.cpp)
```

**Your Environment (please complete the following information):**

 - OS: macOS 10.15.7
 - Compiler: clang 12.0.0
 - PCL Version 1.11.1

**Possible Solution**

This seems like some template instatiation problem, or some implementation header not getting included, but I don't understand it clearly.

--- a/octree/include/pcl/octree/octree_pointcloud_density.h
+++ b/octree/include/pcl/octree/octree_pointcloud_density.h
@@ -145,5 +145,8 @@ public:
 } // namespace octree
 } // namespace pcl
 
+// needed since OctreePointCloud is not instantiated with template parameters used above
+#include <pcl/octree/impl/octree_pointcloud.hpp>
+
 #define PCL_INSTANTIATE_OctreePointCloudDensity(T)                                     \
   template class PCL_EXPORTS pcl::octree::OctreePointCloudDensity<T>;

--- a/octree/include/pcl/octree/octree_pointcloud_singlepoint.h
+++ b/octree/include/pcl/octree/octree_pointcloud_singlepoint.h
@@ -86,5 +86,8 @@ public:
 } // namespace octree
 } // namespace pcl
 
+// needed since OctreePointCloud is not instantiated with template parameters used above
+#include <pcl/octree/impl/octree_pointcloud.hpp>
+
 #define PCL_INSTANTIATE_OctreePointCloudSinglePoint(T)                                 \
   template class PCL_EXPORTS pcl::octree::OctreePointCloudSinglePoint<T>;

